universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
Yes it won't be deleted, but it also can't be safely dereferenced in either thread.  So, it is effectively leaked.  It is as good as deleted, but a lot harder to debug because a dereference could cause a segfault, bad data or no problem at all.  These types of crashes will be more likely to happen in large games than small test cases, because more objects are being accessed.  They will also be architecture specific.

Passing any kind of pointer to another thread must involve consideration of managing access to the dereferenced object.  Using `shared_ptr` is not a way around that work.

Our default instinct should be to use `unique_ptr` for memory management.

If we default to `shared_ptr`, without consideration of why in each case, then we will exchange "technical" resource leaks for "effective" resource leaks.

In this case, we have single resource manager, called `BuildingManager`.  None of the clients store the pointers.  None of the clients are thread safe.   None of the clients' lifetimes exceed the manager, or any of the managed objects.  The pointers should be `unique_ptrs`.